### ISSUE TITLE
LiteratureBuilder: avoid adding empty fields

### ISSUE THREAD

**Issue by jmartinm** [inspirehep/inspire-schemas#135]
```python
In [3]: x = LiteratureBuilder(source='arxiv')

In [4]: x.add_language('')

In [5]: x
Out[5]: LiteratureBuilder(source="arxiv", record={'languages': ['']})
```

**Comment by michamos** [comment]
This is a conscious design decision. @david-caro wanted the buillder to be dumb, @jacquerie to be smart and do validation. I think it would make more sense for the builder to do validation of the subfields, i.e. the builder would contain
```python
def __init__(self, source, record=None):
    # other init
    self.schema = load_schema('hep')

def add_language(...):
    # do whatever it is doing now
    validate(self.record['languages'], self.schema['properties']['languages'])
```

**Comment by jmartinm** [comment]
In principle I was thinking that `filter_empty_parameters` should take care of it (https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/builders.py#L229)

I created this issue because one of the workflows failed because we did (I don't know how):

```python
```

**Comment by michamos** [comment]
that function filters out empty `kwargs`. Here you provided an element of `args`:
```python
In [1]: from inspire_schemas.builders import LiteratureBuilder

In [2]: x = LiteratureBuilder(source='arXiv')

In [3]: x.add_language(language='')

In [4]: x
Out[4]: LiteratureBuilder(source="arXiv", record={})

In [5]: x.add_language('')

In [6]: x
Out[6]: LiteratureBuilder(source="arXiv", record={'languages': ['']})
```

**Comment by michamos** [comment]
It probably is a bug in the builder though. @rikirenz, what do you think?

**Comment by jmartinm** [comment]
Yep, maybe `filter_empty_parameters` should deal with this situation of empty args as well.

**Comment by kaplun** [comment]
What is the conclusion of this issue?

**Comment by rikirenz** [comment]
1. This thing should not happen:
``` Python
builder.add_language('')
```

2. Anyway the idea was this one:

    - When you have an optional parameter you are going to filter it
    - When you have a non optional parameter you are not going to filter it. Who is using the builder should care to filter mandatory parameters

3.  Possible Solutions: 

    - Add an `if` to check if the language is empty and in case we are not going to add the language field.
    - Put `English` as default value for the field language if empty.

**Comment by michamos** [comment]
Discussing IRL with @rikirenz and @jacquerie, we figured out it made more sense to filter everything for now. But maybe the whole approach of the builder should be rethought to be generated automatically from the schema (@jacquerie might explain more).

**Comment by kaplun** [comment]
Automatic generation from the schema will make the builder to change everytime the schema change, thus hindering the very purpose of the builder abstraction. What am I missing?

**Comment by rikirenz** [comment]
We were thinking about a way to create the actual builder directly from the schema. This implies to write a sort of compiler that is able to derive the builder from the schema. It is a cool idea but definitely is not a priority.

**Comment by kaplun** [comment]
You mean a way to kickstart a builder once, and then maintain it by hand, then, correct?

**Comment by rikirenz** [comment]
Well the compiler should be able to derive the builder every time that we do a change. It means that if the compiler is well written we should not touch the builder anymore.

**Comment by rikirenz** [comment]
Anyway before to fix this issue I open this #142. Because I am quite afraid to change the decorator filter without regression tests. For this reason from the moment that this issue it seems not so urgent, I would like to write the tests for the builder before and after fix this.

**Comment by rikirenz** [comment]
depends #142

**Comment by kaplun** [comment]
> Well the compiler should be able to derive the builder every time that we do a change. It means that if the compiler is well written we should not touch the builder anymore.

I am not understanding how the interface of the builder can remain stable if the builder is automatically generated from a changing schema. Anyway since this is not a priority, no need to convince me now :+1:
